File: mosbysample/src/main/java/wangfeixixi/github/com/mosbysample/f/AFragment.java
package wangfeixixi.github.com.mosbysample.f;

import android.support.annotation.NonNull;
import android.util.Log;

import wangfeixixi.github.com.mosbysample.base.BaseF;

/**
 * Created by xuany on 2018/5/9.
 */

public class AFragment extends BaseF<AView, APresent> implements AView {
    private String TAG = "AFragment";

    @Override
    protected int initContentRes() {
        return 0;
    }

    @Override
    protected void initView() {

        tv_name.setText(TAG);
    }

    @Override
    protected void initData(boolean firstLoad, boolean isVisibleToUser) {
        Log.d(TAG, "firstLoad" + firstLoad + "isVisibleToUser" + isVisibleToUser);
        presenter.asdfas();
    }

    @NonNull
    @Override
    public APresent createPresenter() {
        return new APresent();
    }
}

File: mosbysample/src/main/java/wangfeixixi/github/com/mosbysample/f/AFragment.kt
package wangfeixixi.github.com.mosbysample.f

import android.util.Log
import android.view.View
import android.widget.Toast
import butterknife.OnClick
import kotlinx.android.synthetic.main.a_fragment.*
import wangfeixixi.github.com.mosbysample.R

import wangfeixixi.github.com.mosbysample.base.BaseF

class AFragment : BaseF<AView, APresent>(), AView {
    private val TAG = "AFragment"

    override fun initContentRes(): Int = R.layout.a_fragment

    override fun initView() {
        tv_name.text = TAG
    }

    override fun initData(firstLoad: Boolean, isVisibleToUser: Boolean) {
        Log.d(TAG, "firstLoad" + firstLoad + "isVisibleToUser" + isVisibleToUser)
        //        presenter.asdfas();
    }

    override fun createPresenter(): APresent {
        return APresent()
    }

    @OnClick(R.id.tv_name)
    fun tvNameClick(view: View) {
        Log.d(TAG, "测试可以不")
        Toast.makeText(context, "测试可以不", Toast.LENGTH_SHORT).show()
    }
}
